cli/file_fetcher.rs

```diff
@@ -641,140 +634,17 @@ impl FileFetcher {
   }
 }
 
-#[derive(Debug, Eq, PartialEq)]
-enum FetchOnceResult {
```

Comment:
Moved to `HttpClient` so that more of the `reqwest` api usage is buried in there.